File: cifrado/web/codigo/PHP/protected/pages/Cifrado/CifradoInversoSaltos.php
<?php 
class CifradoInversoSaltos{

	private $saltos;

	function __construct($saltos){
		$this->modificarSaltos($saltos);
	}

	function modificarSaltos($nuevoSalto){
		$this->saltos = $nuevoSalto <= 0 ? 1 : $nuevoSalto;
	}

	function cifrar ($textoPlano){
    	$arregloSeparadas = $this->separarPalabra($textoPlano->textoPlano,$this->saltos);
    	$respuesta = "";
    	for ($i=0; $i < count($arregloSeparadas) ; $i++) { 
    		if(strlen($arregloSeparadas[$i]) == $this->saltos ) $respuesta .= strrev($arregloSeparadas[$i]);
    		else $respuesta .= $arregloSeparadas[$i];
    	}
    	return $respuesta;
    }

    private function separarPalabra($texto,$saltos){
    	$respuesta = array();
    	$palabra = "";
    	for ($i=0; $i < strlen($texto) ; $i++) {
    		if($i%$saltos==0 && $i!=0){
    			array_push($respuesta,$palabra);
    			$palabra = $texto[$i];
    		}else $palabra .= $texto[$i];
    	}
    	array_push($respuesta,$palabra);
    	return $respuesta;
    }
	
}
?>

File: cifrado/escritorio/codigo/Ruby/Cifrado/CifradoInversoSaltos.rb
class CifradoInversoSaltos

	attr_reader :saltos

	def initialize saltos
		modificarSaltos saltos
	end

	def modificarSaltos nuevoSalto
		@saltos = nuevoSalto <= 0 ? 1 : nuevoSalto
	end

	def cifrar textoPlano
		arregloSeparadas = separarPalabra(textoPlano.textoPlano, @saltos)
		respuesta = ""
		arregloSeparadas.length.times do |i|
			if arregloSeparadas[i].length == saltos
				respuesta += arregloSeparadas[i].reverse
			else
				respuesta += arregloSeparadas[i]
			end
		end
		respuesta
	end

	private

	def separarPalabra texto, saltos
		respuesta = Array.new
		palabra = ""
		texto.length.times do |i|
			if i%saltos==0 and i != 0
				respuesta.push(palabra)
				palabra=texto[i]
			else
				palabra+=texto[i]
			end
		end
		respuesta.push(palabra)
	end

end